src/server/kernel/wsurface.cpp

```diff
@@ -316,7 +316,7 @@ void WSurface::enterOutput(WOutput *output)
         return;
     wlr_surface_send_enter(d->nativeHandle(), output->handle()->handle());
 
-    connect(output, &WOutput::destroyed, this, [d] {
+    connect(output, &WOutput::aboutToBeInvalidated, this, [d] {
         d->updateOutputs();
```

Comment:
need leave output

Reply:
Yes, especially in multi-screen situations.